### PR TITLE
feat(14507): improve error message for failed txn in activity details view

### LINKED ISSUE #14507: [UX Improvement]: Transaction Failure Message Not Easily Found
### Describe the bug

To see the reason for a failed transaction, you need to open a transaction list item and hover over the word "Failed":
FailedReason

I've been working with hardware wallets for a year and I _just_ found this.   We should present the reason to the user without the need to hover over text.

### Expected final result
https://www.figma.com/design/ZzVQ6iu13C67K807Z1bg5I/Smart-Transactions-1.0?node-id=4296-25303&t=XN3hVQrE3oWM0lCA-1

### Steps to reproduce

1.  Trigger a failed transaction.
2. Hover over word "Failed"

### Error messages or log output

_No response_

### Version

10.13.0

### Build type

_No response_

### Browser

Firefox

### Operating system

MacOS

### Hardware wallet

Trezor

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by DDDDDanica**
## **Description**
When we encounter a failed transaction:
- in activity list from home view, it renders as `failed`, hovering the status will render an error message
- when clicking this activity, the popup view for txn details will render `failed` as well, with no hover effect
- In the meanwhile, there's a new and more user friendly error banner showing for user when txn failed.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/29338?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/14507
Figma: https://www.figma.com/design/ZzVQ6iu13C67K807Z1bg5I/Smart-Transactions-1.0?node-id=4296-25303&t=ff749RbiH6F4IUqk-0

## **Manual testing steps**

1. Render extension and test dapp
2. Trigger a failed txn from test dapp
3. Check activity for that txn
4. Check activity details by clicking item from step 3 and validate the banner, as well as hover

### **Before**
Screenshot 2024-12-18 at 18 25 45

### **After**
Screenshot 2024-12-19 at 01 54 39

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/app/transaction-list-item-details/transaction-list-item-details.component.js [hunk 1]
```diff
@@ -258,22 +279,18 @@ export default class TransactionListItemDetails extends PureComponent {
               data-testid="transaction-list-item-details-tx-status"
             >
               <div>{t('status')}</div>
-              <div>
```
Comment (DDDDDanica) [inline ui/components/app/transaction-list-item-details/transaction-list-item-details.component.js:261]: Clean up
Screenshot 2024-12-18 at 22 31 27

File: ui/components/app/transaction-list-item-details/transaction-list-item-details.component.test.js [hunk 2]
```diff
@@ -71,67 +71,65 @@ const render = async (overrideProps) => {
...
 
-  let result;
-
-  await act(
-    async () =>
-      (result = renderWithProvider(
-        <TransactionListItemDetails {...props} />,
-        mockStore,
-      )),
+  const result = renderWithProvider(
```
Comment (DDDDDanica) [inline ui/components/app/transaction-list-item-details/transaction-list-item-details.component.test.js:82]: There is no asynchronous render here, can be simplified

File: ui/components/app/transaction-status-label/transaction-status-label.js [hunk 3]
```diff
@@ -96,20 +97,29 @@ export default function TransactionStatusLabel({
     }
   }
   ///: END:ONLY_INCLUDE_IF
-
-  return (
+  return shouldShowTooltip ? (
```
Comment (DDDDDanica) [inline ui/components/app/transaction-status-label/transaction-status-label.js:100]: Keep the tooltip for same component here
Screenshot 2024-12-18 at 18 24 09

File: ui/components/app/transaction-list-item-details/index.scss [hunk 4]
```diff
@@ -62,9 +64,10 @@
   }
 
   &__operations {
-    margin: 0 0 16px 16px;
+    margin: 0 16px 16px 16px;
     display: flex;
-    justify-content: flex-end;
+    justify-content: space-between;
```
Comment (DDDDDanica) [inline ui/components/app/transaction-list-item-details/index.scss:69]: When there's a retry button, we need to handle the alignment as well 
Screenshot 2024-12-19 at 03 29 53

File: ui/components/app/transaction-status-label/transaction-status-label.js [hunk 5]
```diff
@@ -96,20 +97,29 @@ export default function TransactionStatusLabel({
...
     <Tooltip
       position="top"
       title={tooltipText}
       wrapperClassName={classnames(
         'transaction-status-label',
-        `transaction-status-label--${statusKey}`,
```
Comment (DDDDDanica) [inline ui/components/app/transaction-status-label/transaction-status-label.js:106]: Duplicated classname
Screenshot 2024-12-19 at 03 36 16

**Comment by pedronfigueiredo** [comment]
Looks good, but would it be worth adding an e2e test?

**Review by pedronfigueiredo — APPROVED** [review]

File: app/_locales/en/messages.json [hunk 6]
```diff
@@ -6404,6 +6404,9 @@
   "transactionFailed": {
     "message": "Transaction Failed"
   },
+  "transactionFailedBannerMessage": {
+    "message": "This transaction would have cost you extra fees, so we stopped it. Your money is still in your wallet."
```
Comment (dbrans) [inline app/_locales/en/messages.json:6408]: Do you think the phrase 'would have cost you extra fees' is clear to users? Even as a developer, I found it a bit ambiguous. Could we consider adding a link to documentation for more context in the future? What do you think, @coreyjanssen?

**Review by dbrans — APPROVED** [review]
Nice work, Danica! Looks like a big improvement.

File: app/_locales/en/messages.json (hunk 6 above)
Comment (DDDDDanica) [reply app/_locales/en/messages.json:6408]: Thanks for sharing your concern Derek, apparently there's a conversation going on with the copy already, you can find in the figma link i shared. It is not obvious actually so I'll share again here 
Screenshot 2024-12-20 at 14 49 52

**Pull request merged**